Controller/RegistrationController.php

```diff
@@ -140,6 +140,23 @@ public function confirmedAction()
 
         return $this->render('FOSUserBundle:Registration:confirmed.html.twig', array(
             'user' => $user,
+            'targetUrl' => $this->getTargetUrlFromSession(),
         ));
     }
+
+    private function getTargetUrlFromSession()
+    {
+        // Set the SecurityContext for Symfony <2.6
+        if (interface_exists('Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface')) {
+            $tokenStorage = $this->get('security.token_storage');
+        } else {
+            $tokenStorage = $this->get('security.context');
+        }
+
+        $key = sprintf('_security.%s.target_path', $tokenStorage->getToken()->getProviderKey());
```

Comment:
`getProviderKey` is not part of `TokenInterface`

A possible alternative (extracted from LogoutUrlGenerator):

```php
            $token = $tokenStorage->getToken();
            if (null !== $token && method_exists($token, 'getProviderKey')) {
                $key = $token->getProviderKey();
            }
```